OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -273,7 +273,7 @@ public ISound Play2DStream(Stream stream, int channels, int sampleBits, int samp
 		public float Volume
 		{
 			get { return volume; }
-			set { AL10.alListenerf(AL10.AL_GAIN, volume = value); }
+			set { AL10.alListenerf(AL10.AL_GAIN, (volume = value) * value * value * value); } // x^4 for linear->exponential
```

Comment:
This should be set up on a decibel scale, such that 0 is unity (no gain) etc. Anyone with a background in audio will be expecting either this or a linear scale.

Reply:
AL_GAIN attribute in OpenAL gets values from 0 to MAX_VAL, where 0 is no sound, 1 is no attenuation (as-is), everything greater than 1 is amplification; on exponential scale [0;1] corresponds to -INFdB to 0dB attenuation, which (x^4) is trying to approximate without overcomplicating the calculations.
https://www.openal.org/documentation/openal-1.1-specification.pdf
Values more than 1.0f were not used here I'd assume because sound amplification is hardware-dependant and could be clamped at arbitrary values, reducing the range of configuration we can apply through slider setting. And we actually don't want that anyway, we just want to allow for lowering the sound volume of the game, as original sound files are expected to be at maximum normalized volume and amplifying that would lead to overdrive and clipping.

Problem with applying f(x) = a*e^(b*x) directly is that you can't then get true zero for f(x) on a scale x = [0, 1] because f(x) only reaches zero at x = -INF.

Check the rationale link (graciously provided by @Phrohdoh I think) .